c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -510,6 +581,46 @@ class DeviceCachingAllocator {
     }
   }
 
+  // CUDAGraph interactions (see Note [Interaction with CUDA graph capture])
+
+  // Called by CUDAGraph::capture_begin
+  void notifyCaptureBegin(CUDACaptureid_t graph_id, CUDACaptureid_t mempool_id) {
+    std::lock_guard<std::recursive_mutex> lock(mutex);
+    captures_underway++;
+    auto it = graph_pools.find(mempool_id);
+    if (it == graph_pools.end()) {
+      graph_pools.insert({mempool_id, PrivatePool()});
+    } else {
+      it->second.use_count++;
+    }
+    // Makes sure this graph_id wasn't somehow assigned a private pool already.
+    TORCH_INTERNAL_ASSERT(capture_to_pool_map.find(graph_id) == capture_to_pool_map.end());
+    capture_to_pool_map[graph_id] = mempool_id;
```

Comment:
You could use insert and look at the returned pair for the assert to skip having to do the lookup twice here